tools/cloner-check/main.go

```diff
@@ -0,0 +1,254 @@
+// Command cloner-check is used to verify that each item stored in the
+// cached_mysql in-memory cache properly implements the fleet.Cloner interface.
+//
+// There are two ways to use this command, the first one is typically used by
+// CI and checks that the cacheable items don't have any changes when compared
+// to the current generated files. That is, if a field is added or modified in
+// a cacheable struct, and those changes haven't been reflected in the
+// generated files yet, it will raise an error, ensuring that the developer
+// takes those changes into account in the custom Clone implementation.
+//
+// The --check flag runs this check mode scenario, but it is optional as running
+// in check mode is the default:
+//
+//	$ go run ./tools/cloner-check/main.go [--check]
+//
+// The second way to use this command is with the --update flag, which is used
+// to update the generated files with the current version of the cacheable
+// items (i.e. the current struct definition). Use this when you've
+// double-checked that the custom Clone implementation is up-to-date and
+// correct.
+//
+//	$ go run ./tools/cloner-check/main.go --update
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/pmezard/go-difflib/difflib"
+)
+
+// If you add a new cacheable struct, make sure to add it to this list.
+var cacheableItems = []fleet.Cloner{
+	&fleet.AppConfig{},
+	&fleet.Pack{},
+	&fleet.ScheduledQuery{},
+	&fleet.Features{},
+	&fleet.TeamMDM{},
+	&fleet.Query{},
+	// TeamAgentOptions is not in the list because it is a json.RawMessage, no fields can change.
+	// Same for ResultCountForQuery, it's just an int.
+}
+
+func main() {
+	flagCheck := flag.Bool("check", false, "Run in check mode (default if no flag is provided)")
+	flagUpdate := flag.Bool("update", false, "Update the generated files with the current cacheable items")
+	flag.Parse()
+
+	// make sure this is run from the root of the repository
+	if _, err := os.Stat(filepath.Join("tools", "cloner-check", "main.go")); os.IsNotExist(err) {
+		fmt.Fprintf(os.Stderr, "error: this command must be run from the root of the repository\n")
+		os.Exit(1)
```

Comment:
just for my own understanding, is there any difference between doing this and `log.Fatalf`? Guessing this is to specify `os.Stderr`?

Reply:
It's just a nicer, controlled output instead of the timestamp and all. I could alter the `log`'s defaults, mind you, but it just seemed more natural to do that.